src/libsyntax/parse/parser.rs

```diff
@@ -5809,13 +5811,19 @@ impl<'a> Parser<'a> {
                                     maybe_append(attrs, extra_attrs));
             return Ok(Some(item));
         }
-        if self.check_keyword(keywords::Unsafe) &&
-            self.look_ahead(1, |t| t.is_keyword(keywords::Impl))
+        if (self.check_keyword(keywords::Unsafe) &&
```

Comment:
For all it is worth, the syntax for functions is like this:

```
[default] [unsafe] fn foo...
```

This probably should match the functions.